### PR TITLE
FB Post Url

### PR DESCRIPTION
**Pull request by hunyoboy**
Added support for list of valid FB post urls with their valid url variations - https://developers.facebook.com/docs/plugins/oembed

### CODE REVIEW

**Comment by hunyoboy** [comment]
@efjacobson please review if you're free. thank you.

File: src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js [hunk 1]
```diff
@@ -1,8 +1,19 @@
 import test from 'tape';
 import getFacebookPostUrl from './getFacebookPostUrl';
 
+const isValidUrl = (str) => {
```
Comment (efjacobson) [inline src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js:4]: @hunyoboy if you want to do this, you would want to put it in the util itself, not in the test only. and you can use the `isValidUrl` util from gdbots/common

**Review by efjacobson — CHANGES_REQUESTED** [review]

File: src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import isValidUrl from '@gdbots/common/isValidUrl';
+
+// supports list of valid Post Urls - https://developers.facebook.com/docs/plugins/oembed
+const getFacebookPostUrl = (str) => {
+  try {
+    const regex = new RegExp('(?:(?:https?:)?\\/\\/)?(?:www\\.)?facebook\\.com\\/([^\\/]+\\/photo(\\.php|s|)(\\/|)|photo(\\.php|s|)(\\/|)|permalink\\.php(\\/|)|media\\/set(\\/|)|questions|notes|[^\\/]+\\/(activity|posts))[\\/?](set=|fbid=|[^\\/]+\\/[^\\/]+\\/|(?=.*\\bstory_fbid=\\b)(?=.*\\bid=\\b).+)?(?:[a-z0-9\\.]+\\/)?([0-9]+)\\/?(?:\\?.[^\\s"]*)?');
+    const postUrlMatches = decodeURIComponent(str).match(regex);
+    return postUrlMatches && isValidUrl(postUrlMatches[0]) ? postUrlMatches[0] : null;
+  } catch (e) {
+    // possible URIError ("malformed URI sequence") exception when used wrongly
```
Comment (efjacobson) [inline src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.js:10]: @hunyoboy do you have an example of something that would throw here?

Comment (hunyoboy) [reply src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.js:10]: @efjacobson also on test! 

https://github.com/triniti/cms-js/blob/c9c808cbbfae86101deb0c97a5498d0691a70f32/src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js#L32

https://github.com/triniti/cms-js/blob/c9c808cbbfae86101deb0c97a5498d0691a70f32/src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.test.js#L29

Comment (efjacobson) [reply src/plugins/blocksmith/components/facebook-post-block-modal/getFacebookPostUrl.js:10]: nice

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 * Instagram embed preview not rendering.
 * Gallery Launch Text not reflecting with red background.
 * Set user node title using first and last name fields in the UserSubscriber to avoid an inconsistency.
+* facebook embed code considered invalid for facebook post block
```
Comment (efjacobson) [inline CHANGELOG.md:10]: @hunyoboy this is being added to the v0.11.0 header

Comment (hunyoboy) [reply CHANGELOG.md]: @efjacobson corrected!

**Review by efjacobson — APPROVED** [review]

**Pull request merged**